crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -470,6 +497,13 @@ impl RecursorDnsHandle {
                 .chain(cached_aaaa.into_iter().flatten())
                 .filter_map(|r| RData::ip_addr(&r))
                 .chain(glue.filter_map(|r| RData::ip_addr(r.data())))
+                .filter(|ip| {
```

Comment:
I'm trying to decide if it's worth moving his filter up before the chain in order to avoid the inner match on the V4/V6. The code is simpler this way, and I suppose that's worth it. Maybe the compiler "does the right thing" in the end.